provider_sources/_compress_providers.py

```diff
@@ -84,131 +84,116 @@ def update_year(provider_or_tile):
-        format = layer["Format"]
-        bounding_lowerleft_corner = layer["ows:WGS84BoundingBox"][
-            "ows:LowerCorner"
-        ]  # given with lon/lat order
-        bounding_upperright_corner = layer["ows:WGS84BoundingBox"][
-            "ows:UpperCorner"
-        ]  # given with lon/lat order
-        lowerleft_corner_lon, lowerleft_corner_lat = bounding_lowerleft_corner.split(
-            " "
-        )
-        upperright_corner_lon, upperright_corner_lat = bounding_upperright_corner.split(
-            " "
-        )
-        bounds = [
-            [float(lowerleft_corner_lat), float(lowerleft_corner_lon)],
-            [float(upperright_corner_lat), float(upperright_corner_lon)],
-        ]
-
-        if format == "application/x-protobuf":
-            pass
-        elif format == "image/x-bil;bits=32":
-            pass
-        elif apikey == "lambert93":
-            pass
-        else:
-            tilelayers_list.append("GeoportailFrance." + name)
-            leaflet["GeoportailFrance"][name] = {
-                "url": """https://wxs.ign.fr/{apikey}/geoportail/wmts?REQUEST=GetTile&SERVICE=WMTS&VERSION=1.0.0&STYLE={style}&TILEMATRIXSET={TileMatrixSet}&FORMAT={format}&LAYER={variant}&TILEMATRIX={z}&TILEROW={y}&TILECOL={x}""",
-                "html_attribution": """<a target="_blank"href="https://www.geoportail.gouv.fr/">Geoportail France</a>""",
-                "attribution": "Geoportail France",
-                "bounds": bounds,
-                "min_zoom": min_zoom,
-                "max_zoom": max_zoom,
-                "apikey": apikey,
-                "format": format,
-                "style": style,
-                "variant": variant,
-                "name": "GeoportailFrance." + name,
-                "TileMatrixSet": TileMatrixSet,
-            }
-
-            possibly_broken = [
-                "Ocsge_Constructions_2002",
-                "Ocsge_Constructions_2014",
-                "Ocsge_Couverture_2002",
-                "Ocsge_Couverture_2014",
-                "Ocsge_Usage_2002",
-                "Ocsge_Usage_2014",
-                "Orthoimagery_Orthophotos_Coast2000",
-                "Pcrs_Lamb93",
-                "Orthoimagery_Ortho_sat_Spot_2013",
-                "Orthoimagery_Orthophotos_1980_1995",
-            ]
-
-            if name in possibly_broken:
-                leaflet["GeoportailFrance"][name]["status"] = "broken"
+
+response = requests.get(ign_wmts_url)
+response_dict = xmltodict.parse(response.content)
+layers_list = response_dict["Capabilities"]["Contents"]["Layer"] # 556 layers
+
+wmts_layers_list = []
+for i in range(len(layers_list)):
+    layer = response_dict["Capabilities"]["Contents"]["Layer"][i]
+    variant = layer.get("ows:Identifier")
+
+    # Rename for better readability
+    name_parts = [part.lower().capitalize().replace("-", "_") for part in variant.split(".")]
+    name = "_".join(name_parts)
+
```

Comment:
This name building should maybe be reviewed to enable the last word of the name to have a capital letter. Maybe a special case should be used for acronyms whatever position it has in the name.